### DISCUSSION TITLE
Possible Memory Leak?

### DISCUSSION THREAD

**Discussion by awcodes** [alpinejs/alpine#790] [4. General]
I may be completely off base on this one, but I've noticed that when working with Alpine and devtools is open the site steadily becomes less and less responsive to alpine events (i.e. clicks, show/hide, etc). As soon as devtools is closed everything goes back to normal.

Is anyone else experiencing this?

Is it even something to worry about since it only happens when devtools is open?

**Comment by KevinBatdorf** [comment]
This is pretty common if you'e using a large set of data in `x-for` as Alpine will rearrange the DOM on every update. Alpine doesn't use a virtual DOM so it's not as speedy for large DOM operations. You might get better performance using `x-show` and rendering on the server. 

> Is it even something to worry about since it only happens when devtools is open?

Personally I wouldn't worry but others might have different opinions.

**Reply by awcodes** [reply]
Thanks for the reply Kevin.

I'm not using any data heavy sets. In fact most of the alpine at the moment is only using x-show on dropdowns and modals.

Could very well just be a chrome thing too. :)

I was mostly just curious if anyone else had noticed/experienced it or not. I've noticed similar thing happening with Livewire too.

Still not a deal breaker. I love Alpine and Livewire.
